basic/mobile_io.py

```diff
@@ -0,0 +1,98 @@
+import hebi
+from time import sleep, time
+
+
+class MobileIO():
+    
+    num_buttons = 8
+    num_sliders = 8
+    current_button_state = [0, 0, 0, 0, 0, 0, 0, 0]
+    current_slider_state = [0, 0, 0, 0, 0, 0, 0, 0]
+    
+    btn_states = ["", "", "", "", "", "", "", ""]
+    
+    
+    def __init__(self, family="HEBI", name="Mobile IO"):
+        # lookup modules on network
+        self.lookup = hebi.Lookup()
+        # wait 2 seconds for list to populate
+        sleep(2)
+        print(self.lookup)
+        self.grp = self.lookup.get_group_from_names([family], [name])
+        print(self.grp)
+        self.cmd = hebi.GroupCommand(self.grp.size)
+        self.fbk = hebi.GroupFeedback(self.grp.size)
+        self.prev_fbk = self.fbk
+        
+    # set the snap position on a slider    
+    def setSnap(self, slider_num, snap_to):
+        self.cmd.io.a.set_float(slider_num, snap_to)
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+
+    # set slider to a value
+    def setAxisValue(self, slider_num, value):
+        self.cmd.io.f.set_float(slider_num, value)
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+
+    # set button mode (momentary/toggle)
+    def setButtonMode(self, button_num, mode):
+        self.cmd.io.b.set_int(button_num, mode)
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+        
+    # set light around button (on/off)
+    def setButtonOutput(self, button_num, output):
+        self.cmd.io.e.set_int(button_num, output)
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+        
+    # set edge led color
+    def setLedColor(self, color): 
+        self.cmd.led.color = color
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+    
+    # send text
+    def setText(self, message):
+        self.cmd.append_log = message
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+    
+    # clear text (currently does not work)
+    def clearText(self): 
+        self.cmd.clear_log = True # bugged needs to be looked at
+        return self.grp.send_command_with_acknowledgement(self.cmd)
+    
+    # send a buzz to the mobile device (only works on iphones)
+    def sendVibrate(self):
+        self.cmd.effort = 1
+        return self.grp.send_command_with_acknowledgement(self.cmd)
```

Comment:
@iamtesch , should we do `send_command_with_ack` here, or just `send_command` ?

Reply:
Hmmm...I would probably stick with `send_command`, unless we are actually doing something with the feedback.

@CalZim , the biggest difference here is that under the covers, `send_command_with_ack` sends a message and then waits for a response.  This means that this call is blocking, and if the return packet is dropped will block until the timeout is reached.

`sendVibrate` is something that is more likely to happen in a control loop as feedback, so the "fire and forget" style of `send_command` is probably more appropriate.

For functions such as `setSnap` which configure the interface as a "one time" action, these are much more appropriate to be done with a `send_command_with_acknowledgement` and have wrapping code in the app to check for success.